### ISSUE TITLE
Changes to `async_` cancelation semantics

### ISSUE THREAD

**Issue by armanbilge** [http4s/blaze#772]
See:
- https://github.com/typelevel/cats-effect/pull/3205

**Comment by armanbilge** [comment]
I let CI try this in https://github.com/armanbilge/blaze/commit/df420f24ce4b0dd9a541f7b8536ca778281a6f36 and there are some test failures.

```
[error] Failed tests:
[error] 	org.http4s.blaze.client.PoolManagerSuite
[error] 	org.http4s.blaze.client.ClientTimeoutSuite
[error] 	org.http4s.blaze.client.BlazeClientSuite
```

**Comment by armanbilge** [comment]
Release candidate is available.
- https://github.com/typelevel/cats-effect/releases/tag/v3.5.0-RC1

**Comment by rossabaker** [comment]
Most of the `async_` calls don't have obvious resources to release.  There's the stage, but shutdown of the stage is probably why we're being canceled.  My two kneejerk reactions are the `Some(F.unit).pure[F]` that's generally discouraged, or shutting down the stage again just in case.  I'm just not sure.

**Comment by armanbilge** [comment]
`Some(F.unit).pure[F]` seems strange, sometimes what I've seen is `F.delay(setupCallback()).as(Some(F.unit))`

**Comment by rossabaker** [comment]
Oh, right.  `async_` hides the delay of setting up the callback.  It should look more like your version if we go that route.

**Comment by rossabaker** [comment]
Addressed in #806
